pandas/core/groupby/groupby.py

```diff
@@ -2890,12 +2891,21 @@ def size(self) -> DataFrame | Series:
         Freq: MS, dtype: int64
         """
         result = self.grouper.size()
+        result_dtype: str | np.dtype = result.dtype
+        if isinstance(self.obj, Series):
```

Comment:
can we just use convert_dtypes for this?  I think you've got another PR doing something similar for another method?

(i actually think for methods that return ints-and-never-NAs it is more useful to return non-nullable dtype, but whatever)

Reply:
> can we just use convert_dtypes for this?

Sure, I will still have to do this original-object introspection though to set the correct `dtype_backend` for `convert_dtype` if that's alright with you

Reply:
either way i guess

Reply:
Used `convert_dtypes` instead